pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaBytesSink.java

```diff
@@ -54,4 +55,11 @@ protected Properties beforeCreateProducer(Properties props) {
     public KeyValue<String, byte[]> extractKeyValue(Record<byte[]> record) {
         return new KeyValue<>(record.getKey().orElse(null), record.getValue());
     }
+
+    @Override
+    public KeyValue<Schema, Schema> extractKeyValueSchemas(Record<byte[]> record) {
+        Schema keySchema = Schema.STRING_SCHEMA;
```

Comment:
any reason why do you use `STRING_SCHEMA`, not `BYTES_SCHEMA`?

Reply:
@sijie This is a type of original key (always String in pulsar).
KafkaProducer does serialization on send with configured serializers.
KafkaSinkWrappingProducer simply passes the key and value as it is to the sink, so the String key is passed as a String.

Reply:
Pulsar already supports `bytes` through `keyBytes`. We should use that instead of using `STRING`.

Reply:
The method accepts `Record<byte[]>` where the `keyBytes` does not exist.

Reply:
I believe that this is a preliminary PR, currently we are only supporting String->byte[]

The next step will be to fully support Schema and KeyValue